Yoast/javascript
# Add support for templates to inner blocks

🚨  **Note**: Test this against https://github.com/Yoast/wordpress-seo/pull/16397

## Summary


This PR can be summarized in the following changelog entry:

* Adds the possibility to define templates for each Innerblocks component.

## Relevant technical choices:

* Made some changes together with @herregroen. These changes will give the possibility to have nested arrays / objects and the combination of both.

## Test instructions

This PR can be tested by following these steps:

* Test this on `wordpress-seo` with https://github.com/Yoast/wordpress-seo/pull/16397. Link the monorepo. In `src/schema-templates/recipe.block.php` add this to line 8:
  ```js
  template=[ [ "yoast/ingredients" ] ]
  ```
  So it will look like this:
  ```js
  {{inner-blocks allowed-blocks=[ "core/paragraph", "core/image", "yoast/ingredients" ] template=[ [ "yoast/ingredients" ] ] appender="button" appenderLabel="Add to recipe" }}
  ```
* Adds the recipe block to your post and see that the ingredients is added to it by default. This is because it defined in the template.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * Post edit -> The recipe block. Has not been released yet.

## UI changes

## Quality assurance


Fixes #
